From c350b7565b666bd5186cb08f196f83e0737347cd Mon Sep 17 00:00:00 2001
From: Vicentiu Galanopulo <vicentiu@balena.io>
Date: Fri, 1 Oct 2021 15:34:35 +0200
Subject: [PATCH] Update gpios pins to reflect etcher-pro hardware
 changes in v2_3

 TFT_RST (GPIO3_IO25, pin142) removed
 TFT_SHLR (GPIO4_IO1, pin192) removed
 TFT_UPDN (GPIO4_IO16, pin194) removed
 TouP_RTS (GPIO4_IO23, pin203) removed
 EE_WP (GPIO3_IO18, pin81) removed
 CEC_STBY (GPIO5_IO29, pin3) removed
 UART1 (GPIO_5_IO22, GPIO_5_IO23, pin69, pin 65) removed
 All HDMI signals removed
 LED_RST (GPIO5_IO4, pin198) moved on (GPIO4_IO23,pin203)
 PWM_FAN (GPIO5_IO4) added. Should be configured as PWM out for the FAN Control.
 For the time can be set it just as a simple High Level output.
 uPCIe_1_RST (GPIO4_IO7, pin139) added. Set it as output and High Level on startup
 SD_WP (GPIO2_IO20, pin67) added.
 INT_IO_EXP (GPIO5_IO28) added
 TFT_BCL_PWM (GPIO5_IO18, pin43), Should be configured as PWM out
                              for the backlight Control.
 For the time can be set it just as a simple High Level output.
 PWM_LED (GPIO5_IO3), set to low
 RST_LED (GPIO4_IO23), set to high

Upstream-status: Inappropriate [configuration]
Signed-off-by: Vicentiu Galanopulo <vicentiu@balena.io>
---
 board/compulab/cl-som-imx8/cl-som-imx8.c | 17 ++++++++++-------
 1 file changed, 10 insertions(+), 7 deletions(-)

diff --git a/board/compulab/cl-som-imx8/cl-som-imx8.c b/board/compulab/cl-som-imx8/cl-som-imx8.c
index c0cd668547..248000a23c 100644
--- a/board/compulab/cl-som-imx8/cl-som-imx8.c
+++ b/board/compulab/cl-som-imx8/cl-som-imx8.c
@@ -338,16 +338,19 @@ int board_init(void)
 	gpio_direction_output(IMX_GPIO_NR(4, 5), 1);
 
 	gpio_request(IMX_GPIO_NR(5, 18),"tft_bcl_pwm");
-	gpio_direction_output(IMX_GPIO_NR(5, 18), 0);
+	gpio_direction_output(IMX_GPIO_NR(5, 18), 1);
 
-	gpio_request(IMX_GPIO_NR(4, 1),"tft_shlr");
-	gpio_direction_output(IMX_GPIO_NR(4, 1), 0);
+	gpio_request(IMX_GPIO_NR(5, 4),"pwm_fan");
+	gpio_direction_output(IMX_GPIO_NR(5, 4), 1);
 
-	gpio_request(IMX_GPIO_NR(4, 16),"tft_updw");
-	gpio_direction_output(IMX_GPIO_NR(4, 16), 1);
+	gpio_request(IMX_GPIO_NR(4, 7),"uPCIe_1_RST");
+	gpio_direction_output(IMX_GPIO_NR(4, 7), 1);
 
-	gpio_request(IMX_GPIO_NR(3, 25),"tft_rst");
-	gpio_direction_output(IMX_GPIO_NR(3, 25), 1);
+	gpio_request(IMX_GPIO_NR(4, 23),"reset_led");
+	gpio_direction_output(IMX_GPIO_NR(4, 23), 1);
+
+	gpio_request(IMX_GPIO_NR(5, 3),"led_outenable");
+	gpio_direction_output(IMX_GPIO_NR(5, 3), 0);
 
 #if defined(CONFIG_USB_DWC3) || defined(CONFIG_USB_XHCI_IMX8M)
 	init_usb_clk();
-- 
2.17.1

